src/ica_api.c

```diff
@@ -1242,6 +1242,28 @@ int ica_ecdsa_verify(ica_adapter_handle_t adapter_handle,
 	return rc;
 }
 
+int ica_ec_key_get_public_key(ICA_EC_KEY *key, unsigned char *q, unsigned int *q_len)
+{
+    if (!key || !(key->X) || privlen_from_nid(key->nid) < 0)
+	return EINVAL;
+
+    memcpy(q, key->X, 2*privlen_from_nid(key->nid));
+    *q_len = 2*privlen_from_nid(key->nid);
+
+    return 0;
+}
+
+int ica_ec_key_get_private_key(ICA_EC_KEY *key, unsigned char *d, unsigned int *d_len)
```

Comment:
can key parameter be qualified by const? it seems to be read only in this context.

Reply:
Yep, same as for get_public_key